### PR TITLE
Expanded unit testing, modernized ModRamFunctions, KpMax bugfix

### LINKED ISSUE #97: ModRamFunctions needs legacy control structure updating
While we can't necessarily control some of our dependencies, there are still a couple of GOTO statements in the codebase (`ModRamFunctions.f90`). These are coupled with `DO [line number] ... CONTINUE` loop control, and can be rewritten to conform to modern Fortran standards.

These are an ideal case for some unit tests. So first we can write unit tests for functions with old-style control flow (and make sure that the tests are thorough and pass), then update the functions to use modern approaches.

### PR DESCRIPTION
**Pull request by drsteve**
ModRamFunctions has been pared down by removing the set of functions required by the old `ERFF` routine, now replaced by the F2008 intrinsic `erf`.

The unit tests for `ERFF` have also been removed. `Gcoul` is still tested, as one of the primary uses of `erf`.

Unit tests were added for `NewtFit` and `update_indices`.

As the new `update_indices` tests were failing, the behavior of the indexing on finding `Kpmax12` and `Kpmax24`was investigated. This has been corrected so indices are supplied in forward order. Under reverse order the `Kpmax` variables always took the value of Kp for the given time.

~~Last item, which is TBD (and when complete will move this PR out of draft) is a transition of RAM-SCB functions to use the new _explicit_ return style.~~

Closes #97 and partially addresses #90

### CODE REVIEW

**Comment by drsteve** [comment]
Tested this locally and with the GHA test runner for:
- d1f784e: passes regression tests, but fails the unit test for Kpmax12/24
- f0cb678: passes the unit test for Kpmax12/24 but fails the regression tests

I'm a little confused by the regression test failure as those tests don't use the plasmasphere...

**Comment by drsteve** [comment]
I've pulled back the scope of this PR so it's ready to go now. If I can get @Pheosics to take a quick look over it and leave any comments that'd be helpful, just so it's not only my eyeballs on this code before merge...

Updating the function calls is largely cosmetic and can come basically any time, so I'm happy to leave that out for now.

File: src/ModRamFunctions.f90 [hunk 1]
```diff
@@ -723,4 +510,50 @@ subroutine test_Gcoul(verbose)
 
   end subroutine test_Gcoul
 
+  subroutine test_NewtFit(verbose)
```
Comment (Pheosics) [inline src/ModRamFunctions.f90:513]: I'm wondering if we should just get rid of the NewtFit functions completely and drop this test. With the GSL interpolation functions we don't actually use this anywhere.

**Comment by Pheosics** [comment]
Commits look good, besides the question regarding the NewtFit and NewtFitLarge functions and tests I think it is ready to be merged. If you would like I can merge, otherwise I am signing off on this.

File: src/ModRamFunctions.f90 (hunk 1 above)
Comment (drsteve) [reply src/ModRamFunctions.f90:513]: I'm happy to add paring down the codebase to drop obsolete or redundant code as a task for the next release (or maybe the one after, as I want to get to doing numbered versions more frequently). For now this doesn't really impact anything beyond showing that a function we don't use any more has expected behavior...

**Comment by drsteve** [comment]
If you're happy - including leaving the NewtFit stuff for now but slating it for removal later - then can I ask you to merge? That way it's not me merging my own PRs, which always feels a little weird.

**Pull request merged**